cpmpy/transformations/linearize.py

```diff
@@ -119,6 +119,21 @@ def linearize_constraint(lst_of_expr, supported={"sum","wsum"}, reified=False):
                     lhs = Operator("wsum",[[lhs.args[0]], [lhs.args[1]]])
                     cpm_expr = eval_comparison(cpm_expr.name, lhs, rhs)
 
+                elif lhs.name == "pow" and "pow" not in supported:
+                    if "mul" not in supported:
+                        raise NotImplementedError("Cannot linearize power without multiplication")
+                    if not is_num(lhs.args[1]):
+                        raise NotImplementedError("Cannot linearize power with ")
+                    # only `POW(b,n) == IV` supported, with n being an integer, post as b*b*...*b (n times) == IV
+                    x, n = lhs.args
+                    new_lhs = 1
+                    for exp in range(n):
+                        new_lhs, new_cons = get_or_make_var(x * new_lhs)
+                        newlist.extend(new_cons) # TODO: when reified, this will make again new vars - should post directly toplevel instead
```

Comment:
what can of reification can make it to linearization? we already flattened?

Reply:
Hmm good point, Indeed the TODO will not actually occur in practice as we first do `reify_rewrite` in the interfaces.
I was thinking of the case where `b -> x ** 3` is first linearized, and then the linearized power is put toplevel if the power is not supported in a reifiied context, but that won't be the case in practice :)